MainApp/Entities/Controllers/ShiftController.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using MainApp.Entities.Models;
+using MainApp.Entities.Presenters;
+
+namespace MainApp.Entities.Controllers
+{
+    public class ShiftController
+    {
+        public IReadOnlyList<HeroPresenter> HeroPresenters => _heroPresenters;
+        public MovementDeckController MovementDeckController => _movementDeckControllers[_currentHeroIndex];
+
+        private List<HeroModel> _heroModels;
+        private List<HeroController> _heroControllers;
+        private List<HeroPresenter> _heroPresenters;
+        private int _currentHeroIndex = 0;
+
+        private readonly int _countOfPlayers;
+        private MovementDeckPresenter _movementDeckPresenter;
+
+        private GameFieldController _gameFieldController;
+
+        private List<MovementDeckController> _movementDeckControllers;
+
+        private int _currentHeroSuccessfulMoves = 0;
```

Comment:
I am confused by the title of the variable: it would be better, imho, if the title was _currentHeroSuccessfulMovesCount.

Reply:
Thats a good catch, will rename for proper readability

Reply:
Done